clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -278,6 +284,127 @@ static mlir::Value emitX86MaskTest(CIRGenBuilderTy &builder, mlir::Location loc,
   return emitIntrinsicCallOp(builder, loc, intrinsicName, resTy,
                              mlir::ValueRange{lhsVec, rhsVec});
 }
+// TODO: The cgf parameter should be removed when all the NYI cases are
+// implemented.
+static std::optional<mlir::Value> emitX86MaskedCompareResult(CIRGenFunction &cgf, CIRGenBuilderTy &builder,
+                                              mlir::Value cmp, unsigned numElts,
+                                              mlir::Value maskIn,
+                                              mlir::Location loc) {
+  if (maskIn) {
+    cgf.cgm.errorNYI(loc, "emitX86MaskedCompareResult");
```

Comment:
```suggestion
    cgf.cgm.errorNYI(loc, "emitX86MaskedCompareResult");
    return {};
```
Returning a null `mlir::Value` rather than `std::nullopt` indicates that we already reported an error.